src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py

```diff
@@ -200,6 +210,9 @@ def collectAgentInfo(self):
             if float(disk['percent'].strip('%')) >= diskUseThreshold and \
                             disk['mounted'] not in self.config.AnalyticsDataCollector.ignoreDisk:
                 agentInfo['disk_warning'].append(disk)
+                self.config.WorkQueueManager.queueParams['DrainMode'] = True
+                if self.config_path is not None and not isDrainMode(self.config):
```

Comment:
If we have more than 1 disk warning, then we'd be saving the config > 1 times. I suggest keeping this if after this for loop. See changes in the file I'm going to share.